thunder/executors/nvfuserex_impl.py

```diff
@@ -2755,56 +2755,56 @@ def embedding(
-#     if not enable_scatter:
-#         return False
-#
-#     # TODO: limited support inside nvfuser. remove this when codegen support is generalized.
-#     if len(indices) != 1 or indices[0].ndim != 1:
-#         return False
-#
-#     if accumulate == True:
-#         return False
-#
-#     return True
-#
-#
-# def index_put(
-#     a: TensorProxy,
-#     /,
-#     indices: Sequence[TensorProxy],
-#     values: TensorProxy,
-#     accumulate: bool,
-#     *,
-#     fd: FusionDefinition,
-#     lc_to_nv_map: dict,
-# ) -> any:
-#     utils.check(
-#         not accumulate, lambda: "Unsupported accumulate in index_put by nvfuserex", exception_type=AssertionError
-#     )
-#     nva = getnv(a, fd, lc_to_nv_map)
-#     nvi = getnv(indices[0], fd, lc_to_nv_map)
-#     # construct the shape of broadcast indices tensor as
-#     # [-1, *a.shape[1:]]
-#     shapes = nva.shape()
-#     flag = [-1]
-#     for i in range(1, nva.ndim):
-#         flag += [shapes[i]]
-#     # broadcast index tensor nvi to abide to scatter semantics
-#     nvi_b = fd.ops.broadcast_in_dim(nvi, flag, [0])
-#
-#     nvs = getnv(values, fd, lc_to_nv_map)
-#
-#     # index_put is translated to scatter in nvfuser
-#     return fd.ops.scatter(nva, nvi_b, nvs, 0)
-#
-#
-# register_supported(PrimIDs.INDEX_PUT, index_put, _index_put_check)
+def _index_put_check(a: TensorProxy, /, indices: Sequence[TensorProxy], values: TensorProxy, accumulate: bool) -> bool:
+    # temporary flag to allow scatter-like operations to be consumed by nvfuserex
+    enable_scatter: None | bool = get_compile_option("nv_enable_scatter", "Enable nvFuser scatter-like operations.")
+    if not enable_scatter:
+        return False
+
+    # TODO: limited support inside nvfuser. remove this when codegen support is generalized.
+    if len(indices) != 1 or indices[0].ndim != 1:
+        return False
+
+    if accumulate == True:
+        return False
+
+    return True
+
+
+def index_put(
+    a: TensorProxy,
+    /,
+    indices: Sequence[TensorProxy],
+    values: TensorProxy,
+    accumulate: bool,
+    *,
+    fd: FusionDefinition,
+    lc_to_nv_map: dict,
+) -> any:
+    utils.check(
+        not accumulate, lambda: "Unsupported accumulate in index_put by nvfuserex", exception_type=AssertionError
+    )
+    nva = getnv(a, fd, lc_to_nv_map)
+    nvi = getnv(indices[0], fd, lc_to_nv_map)
+    # construct the shape of broadcast indices tensor as
+    # [-1, *a.shape[1:]]
+    shapes = nva.shape()
+    flag = [-1]
+    for i in range(1, nva.ndim):
+        flag += [shapes[i]]
+    # broadcast index tensor nvi to abide to scatter semantics
+    nvi_b = fd.ops.broadcast_in_dim(nvi, flag, [0])
+
+    nvs = getnv(values, fd, lc_to_nv_map)
+
+    # index_put is translated to scatter in nvfuser
+    return fd.ops.scatter(nva, nvi_b, nvs, 0)
+
+
+register_supported(PrimIDs.INDEX_PUT, index_put, _index_put_check)
 
 
-def _cross_entropy_check_(
+def _cross_entropy_check(
```

Comment:
it'd be more desirable to put this change in a different PR
```suggestion
def _cross_entropy_check_(
```